lenskit-core/src/main/java/org/lenskit/data/entities/EntityBuilder.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * LensKit, an open source recommender systems toolkit.
+ * Copyright 2010-2014 LensKit Contributors.  See CONTRIBUTORS.md.
+ * Work on LensKit has been funded by the National Science Foundation under
+ * grants IIS 05-34939, 08-08692, 08-12148, and 10-17697.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package org.lenskit.data.entities;
+
+/**
+ * Base class for entity builders.  The standard implementation is {@link BasicEntityBuilder}, which can be created
```

Comment:
I'm trying to standardize away from calling providers builders. Still need to make this reality in the LensKit code, but going forward, builders are *not* providers.

Builders are used for verbose, incremental construction of (generally immutable) objects. In some cases, such as `EntityBuilder`, they can be used to give other code the ability to build arbitrary types of objects (for example, `RatingBuilder` will implement `EntityBuilder`, so the same CSV-parsing code can build `Rating` objects and `BasicEntity` and `LikeCount` and whatever).